### PR TITLE
Run Jenkinsfile in parallel

### PR DESCRIPTION
**Pull request by aelred**
Build once, then run everything in parallel

### CODE REVIEW

**Comment by burukuru** [comment]
!rtg

**Comment by aelred** [comment]
!rtg - random failure, logged [here](https://work.grakn.ai/entity/18086)

**Comment by aelred** [comment]
yay! @fppt have a look whenever

File: Jenkinsfile [hunk 1]
```diff
@@ -97,45 +93,72 @@ def buildGrakn() {
...
+        }
     }
-}
 
-// Deploy long-running instance on stable branch
-if (env.BRANCH_NAME == 'stable') {
-    node {
-        String workspace = pwd()
-        checkout scm
+    jobs = [
```
Comment (burukuru) [inline Jenkinsfile]: Could you add some comments here to say what you're doing?

File: Jenkinsfile [hunk 2]
```diff
@@ -97,45 +93,72 @@ def buildGrakn() {
...
+                    }
+                }
+            }
+        }
+    ];
 
-        stage('Build Grakn') {
-            withScripts(workspace) {
-                buildGrakn()
+    for (String moduleName : integrationTests) {
```
Comment (burukuru) [inline Jenkinsfile]: and here

File: Jenkinsfile [hunk 3]
```diff
@@ -97,45 +93,72 @@ def buildGrakn() {
...
             }
         }
+    }
 
-        stage('Deploy Grakn') {
-            sshagent(credentials: ['jenkins-aws-ssh']) {
-                sh "scp -o StrictHostKeyChecking=no grakn-dist/target/grakn-dist*.tar.gz ubuntu@${LONG_RUNNING_INSTANCE_ADDRESS}:~/"
-                sh "scp -o StrictHostKeyChecking=no scripts/repeat-query ubuntu@${LONG_RUNNING_INSTANCE_ADDRESS}:~/"
-                ssh "'bash -s' < scripts/start-long-running-instance.sh"
+    parallel(jobs);
```
Comment (burukuru) [inline Jenkinsfile]: and here

File: Jenkinsfile [hunk 4]
```diff
@@ -97,45 +93,72 @@ def buildGrakn() {
...
+
+    node {
```
Comment (burukuru) [inline Jenkinsfile]: You can remove it as there's a nested 'node' clause here.

Comment (aelred) [reply Jenkinsfile]: This node is actually not inside another. I think the `parallel` part has to be executed outside of a node, so I leave the first node to run then - then I have to make a new one to deploy at the end.

Comment (burukuru) [reply Jenkinsfile]: There's another node clause on 153 though.

Comment (aelred) [reply Jenkinsfile]: Ohhh! I see it, thanks.

**Pull request merged**